### PR TITLE
post_message migration [Issue #772]

### PR DESCRIPTION
**Pull request by Andy-Waine**
Resolves: Issue https://github.com/hyperledger/aries-vcx/issues/772 - Move out post_message to shared_vcx

##Short summary
-Migrated httpclient.rs file to shared_vcx directory
-The issue requested that the post_message() function be moved, but there were other items within httpclient.rs that post_message() depended upon.
-aries-vcx is successfully compiling with the changes provided.

### CODE REVIEW

File: agency_client/src/internal/messaging.rs139
Comment (Patrik-Stas) [inline agency_client/src/internal/messaging.rs139:1]: seems like this file was created by accident?

File: shared_vcx/src/errors/error.rs [hunk 1]
```diff
@@ -53,3 +53,85 @@ where
 }
 
 pub type SharedVcxResult<T> = Result<T, SharedVcxError>;
+
+pub mod prelude {
+    pub use crate::errors::error::*;
+}
+
+#[derive(Copy, Clone, Eq, PartialEq, Debug, thiserror::Error)]
+pub enum SharedAgencyClientErrorKind {
```
Comment (Patrik-Stas) [inline shared_vcx/src/errors/error.rs:62]: There already is `SharedVcxErrorKind` under `shared_vcx/src/errors/error.rs`, just extend that one please

Comment (Patrik-Stas) [reply shared_vcx/src/errors/error.rs:62]: Please also check if all of these added error kinds are actually being used.

File: shared_vcx/src/testing/mocking.rs
Comment (Patrik-Stas) [inline shared_vcx/src/testing/mocking.rs:1]: It seems this is copied from `agency_client` - can the original `mocking.rs` be deleted now?

**Review by Patrik-Stas — CHANGES_REQUESTED** [review]
Left few comments, also note that some testing jobs are failing. Take a look at which jobs are failing, what test command they run in order to reproduce the issues locally.

**Comment by Patrik-Stas** [comment]
One more recommendation - when you are working on a pull request, it's better when you create and working within a custom feature branch. 
It's then easier 
- for maintainers of target repository checkout your work locally and take a better look at changes
- also for you, because since in your forked repository you pushed commits to `main`, it will have conflicts with `hyperledger/aries-vcx` and you won't be able to sync it.

**Comment by Andy-Waine** [comment]
Hey @Patrik-Stas thank you for the recommendations, I will work on those next. In regards to cutting a feature branch, I attempted that initially, but when I tried to push the branch to the `aries-vcx` repo, I received an error stating that I did not have the proper permissions to push, leading me to fork. I'll look into it some more and try again.

Thanks!

**Comment by Patrik-Stas** [comment]
> Hey @Patrik-Stas thank you for the recommendations, I will work on those next. In regards to cutting a feature branch, I attempted that initially, but when I tried to push the branch to the `aries-vcx` repo, I received an error stating that I did not have the proper permissions to push, leading me to fork. I'll look into it some more and try again.
> 
> Thanks!

Yes, pushing branch directly under Hyperledger repo won't work. The recommendation is that even though it's your own fork, it's better to work on a feature branch within your fork. And then it's easy to keep your `main` branch in sync with the origin (hyperledger) `main` branch

**Comment by Andy-Waine** [comment]
> Yes, pushing branch directly under Hyperledger repo won't work. The recommendation is that even though it's your own fork, it's better to work on a feature branch within your fork. And then it's easy to keep your `main` branch in sync with the origin (hyperledger) `main` branch

I understand now. Thank you for the clarification, @Patrik-Stas!

**Pull request closed without merging**